tests/general/testdata/incompat_env_config_source_labels_expected.yaml

```diff
@@ -30,13 +30,13 @@ resourceMetrics:
                         stringValue: prefix-$$$$an-envvar-value
                     - key: octuple-dollar
                       value:
-                        stringValue: $$$an-envvar-value-suffix
+                        stringValue: $$$${env:AN_ENVVAR}-suffix
```

Comment:
Apparently, we have been expanding/resolving such strings twice after https://github.com/signalfx/splunk-otel-collector/pull/2893. And this test validates wrong behavior. The issue isn't fixed by this PR. However, https://github.com/open-telemetry/opentelemetry-collector/pull/10560 blocks the second expansion if we have an even number of $ after the first expansion. 

I'm updating the tests because it's a step in the right direction. In the meantime, I'm looking into the bigger problem.

Reply:
Good work finding this.